src/pointer.rs

```diff
@@ -1184,6 +1188,56 @@ const fn validate(value: &str) -> Result<&str, ParseError> {
     Ok(value)
 }
 
+/*
+░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░
+╔══════════════════════════════════════════════════════════════════════════════╗
+║                                                                              ║
+║                              TopicalParseError                               ║
+║                             ¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯                              ║
+╚══════════════════════════════════════════════════════════════════════════════╝
+░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░
+*/
+#[derive(Debug, PartialEq)]
+pub struct ParseBufError {
+    pub value: String,
```

Comment:
So we're not going with `Cow` here? I'm fine with it, but all the more reason to make it private so we have the option to change in the future without another hard break.

Reply:
If we do `Cow` here, I guess we can consolidate them back down to `ParseError`.

The thinking was that this is only created from `PointerBuf::parse`, which has a `String`, or from a method call (needs renaming) on `ParseError` where you provide the source. 

You're far more knowledgeable on rust and conventions than I - if you think ditching `ParseBufErrror` (or whatever it ends up being called) in favor of throwing a `Cow` on `ParseError`'s variants, we can go that route.

Reply:
It's not a big deal since this is the uncommon path, but my thinking is that with a `Cow` we can avoid one allocation in `PointerBuf::parse` even if we do fail validation. We need another (very slightly breaking) change to the API to take advantage of it:

```rust
pub fn parse(s: impl Into<Cow<'_, str>>) -> Result<Self, ParseBufError> {
    let s = s.into();
    validate(&s).map_err(|err| err.with_subject(s))?;
    Ok(Self(s.into_owned()))
}
```